src/helics/application_api/FederateInfo.cpp

```diff
@@ -225,113 +216,136 @@ int getOptionIndex (std::string val)
     return -1;
 }
 
-void FederateInfo::loadInfoFromArgs (int argc, const char *const *argv)
+using namespace std::string_literals;
+
+std::unique_ptr<helicsCLI11App> FederateInfo::makeCLIApp ()
 {
-    variable_map vm;
-    auto res = argumentParser (argc, argv, vm, InfoArgs);
-    if (res == versionReturn)
-    {
-        std::cout << helics::versionString << '\n';
-    }
-    if (res < 0)
-    {
-        return;
-    }
-    if (vm.count ("name") > 0)
-    {
-        defName = vm["name"].as<std::string> ();
-    }
-    if (vm.count ("corename") > 0)
-    {
-        coreName = vm["corename"].as<std::string> ();
-    }
-    if (vm.count ("core") > 0)
-    {
-        coreType = helics::coreTypeFromString (vm["core"].as<std::string> ());
-    }
-    if (vm.count ("type") > 0)
-    {
-        coreType = helics::coreTypeFromString (vm["type"].as<std::string> ());
-    }
-    if (vm.count ("coretype") > 0)
-    {
-        coreType = helics::coreTypeFromString (vm["coretype"].as<std::string> ());
-    }
+    auto app = std::make_unique<helicsCLI11App> ("Federate Info Parsing");
+
+    app->add_option ("--name,-n", defName, "name of the federate");
+    app->add_option ("--corename", coreName, "the name of the core to create or find")->ignore_underscore ();
+    app->addTypeOption ();
+    app->add_option ("--coreinitstring,-i,--coreinit", coreInitString, "The initialization arguments for the core")
+      ->transform ([] (std::string arg) {
+          arg.insert (arg.begin (), ' ');
+          return arg;
+      })
+      ->ignore_underscore ();
+    app->add_option ("--broker,--broker_address", broker, "address or name of the broker to connect")
```

Comment:
We should remove the underscore here, so that it is consistent with the other arguments.